spikeinterface/toolkit/preprocessing/filter.py

```diff
@@ -63,33 +64,42 @@ def __init__(self, recording, band=[300., 6000.], btype='bandpass',
         # self.coeff is 'sos' or 'ab' style
         coeff = scipy.signal.iirfilter(N, Wn, analog=False, btype=btype, ftype=ftype, output=filter_mode)
 
-        BasePreprocessor.__init__(self, recording, dtype=dtype)
-        dtype_base = self.get_dtype()
+        if dtype is None:
+            dtype_base = recording.get_dtype()
+        else:
+            dtype_base = dtype
+        dtype_base, force_traces_dtype = _correct_uint(dtype_base)
+
+        BasePreprocessor.__init__(self, recording, dtype=dtype_base)
```

Comment:
I think it should be force_traces_dtype.
Because the dtype of the new recording will be int and not uint.